jupyterhub/scopes.py

```diff
@@ -18,7 +18,18 @@ def get_scopes_for(orm_object):
     scopes = set()
     if orm_object is None:
         return scopes
-    elif isinstance(orm_object, orm.APIToken):
+
+    if not isinstance(orm_object, orm.Base):
+        from .user import User
+
+        if isinstance(orm_object, User):
+            orm_object = orm_object.orm_user
```

Comment:
This is the only place we need to handle the User wrapper, and we immediately cast it to the orm object so everything else gets the lower-level objects